application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = (int) Math.ceil(pendingReminders.size() / (double) MAX_PAGE_LENGTH);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
+
+        if (pendingReminders.isEmpty()) {
+            remindersEmbed.setDescription("No pending reminders");
+        } else {
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: " + pageToShow + "/" + totalPages);
+
+                getRemindersPage(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+            }
+        }
+
+        return listMessage.addEmbeds(remindersEmbed.build()).build();
+    }
+
+    private List<Button> createPageTurnButtons(int pageNumber, int totalPages) {
+        int previousButtonTurnPageBy = -1;
+        Button previousButton =
+                createPageTurnButton(PREVIOUS_BUTTON_LABEL, pageNumber, previousButtonTurnPageBy);
+        if (pageNumber <= 1) {
+            previousButton = previousButton.asDisabled();
+        }
+
+        int nextButtonTurnPageBy = 1;
+        Button nextButton =
+                createPageTurnButton(NEXT_BUTTON_LABEL, pageNumber, nextButtonTurnPageBy);
+        if (pageNumber >= totalPages) {
+            nextButton = nextButton.asDisabled();
+        }
+
+        return List.of(previousButton, nextButton);
+    }
+
+    private Button createPageTurnButton(String label, long pageNumber, int turnPageBy) {
+        return Button.primary(
+                generateComponentId(String.valueOf(pageNumber), String.valueOf(turnPageBy)), label);
+    }
+
+    private static List<PendingRemindersRecord> getRemindersPage(
+            Result<PendingRemindersRecord> remindersRecords, int pageNumber) {
+        List<PendingRemindersRecord> reminders = new ArrayList<>(MAX_PAGE_LENGTH);
+
+        int start = (pageNumber - 1) * MAX_PAGE_LENGTH;
+        int end = Math.min(start + MAX_PAGE_LENGTH, remindersRecords.size());
+
+        for (int i = start; i < end; i++) {
+            reminders.add(remindersRecords.get(i));
+        }
+
+        return reminders;
+    }
+
+    private static void addReminderAsField(PendingRemindersRecord reminder, EmbedBuilder embed) {
+        BiFunction<Long, Instant, String> getDescription = (channelId, remindAt) -> """
+                Channel: %s
+                Remind at: %s""".formatted(MessageUtils.mentionChannelById(channelId),
+                TimeFormat.DEFAULT.format(remindAt));
```

Comment:
Since this does not access any local variables you can move this to its own method.

Reply:
i can but i won't, i dont like having a method for every little thing

i can even just make it a normal string and format it in the stream thing

Reply:
But what you are doing here is basically creating a method in a method. So why dont you just separate them.

Reply:
yea i agree, changing it to a string

Reply:
coz this is a proper use of functional interface

Reply:
I would go with the String approach. It is only used a single time anyways

Reply:
its not like i just put the channelId and remindAt,, they are not even strings i use some methods against them and then put them in string.

just let it be how it is

Reply:
Look here it does exactly the same, except you dont need your BiFunction. If your method used this BiFunction multiple times and used some things that are the same for all descriptions, then it would totally make sense. This does not however.

```java
private static void addReminderAsField(PendingRemindersRecord reminder, EmbedBuilder embed) {
        String content = reminder.getContent();
        long channelId = reminder.getChannelId();
        Instant remindAt = reminder.getRemindAt();

        String description = """
                Channel: %s
                Remind at: %s""".formatted(MessageUtils.mentionChannelById(channelId),
                TimeFormat.DEFAULT.format(remindAt));

        embed.addField(MessageUtils.abbreviate(content, MAX_REMINDER_TITLE_LENGTH), description,
                false);
    }
```

Reply:
yes good idea